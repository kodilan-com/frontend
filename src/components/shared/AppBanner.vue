<script>
import queryUtils from '../../utils/query';
import LocationSelect from './LocationSelect';

export default {
  components: {
    LocationSelect,
  },
  data() {
    return {
      params: {
        query: '',
        location: '',
      },
    };
  },
  methods: {
    handleSearch() {
      const params = queryUtils.getParams(this.params);

      if (params) {
        this.$router.push({ path: 'ilan-ara', query: params });
      }
    },
  },
};
</script>

<template>
  <div id="banner">
    <div class="container">
      <div class="sixteen columns">
        <div class="search-container">
          <h2>İlan Ara</h2>
          <input
            v-model="params.query"
            @keypress.enter="handleSearch"
            type="text"
            class="ico-01"
            placeholder="Pozisyon adı, teknoloji adı"
          >
          <div class="banner--location-select">
            <location-select
              v-model="params.location"
              :value="params.location"
              :searchable="true"
            />
          </div>
          <button @click="handleSearch" class="tag-search-btn">
            <i class="fa fa-search tag-search-btn" />
          </button>
          <div class="clearfix" />
          <div class="browse-jobs">
            <router-link to="/etiket/frontend" class="tag-header-tags">
              Frontend
            </router-link>
            <router-link to="/etiket/backend" class="tag-header-tags">
              Backend
            </router-link>
            <router-link to="/etiket/mobile" class="tag-header-tags">
              Mobile
            </router-link>
            <router-link to="/etiket/devops" class="tag-header-tags">
              DevOps
            </router-link>
          </div>
          <div class="browse-jobs next-line">
            <router-link to="/ilan-ara?location=istanbul" class="tag-header-tags">
              İstanbul
            </router-link>
            <router-link to="/ilan-ara?location=izmir" class="tag-header-tags">
              İzmir
            </router-link>
            <router-link to="/ilan-ara?location=ankara" class="tag-header-tags">
              Ankara
            </router-link>
            <router-link to="/ilan-ara?location=Remote" class="tag-header-tags">
              Remote
            </router-link>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>
