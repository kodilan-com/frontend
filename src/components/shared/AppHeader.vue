<script>
import { mapState, mapActions } from 'vuex';
import AppBanner from './AppBanner';
import AppTopbar from './AppTopbar';

export default {
  props: {
    withBanner: {
      type: Boolean,
      required: false,
      default: false,
    },
  },
  components: {
    AppBanner,
    AppTopbar,
  },
  computed: {
    ...mapState(['isLoggedIn']),
  },
  methods: {
    ...mapActions(['logout']),
  },
};
</script>

<template>
  <section>
    <app-topbar />
    <header class="app-header sticky-header">
      <div class="container">
        <div class="sixteen columns">
          <div id="logo">
            <h2>
              <router-link to="/">
                <span>{ </span>kod<span>, </span>ilan<span> }</span>
              </router-link>
            </h2>
          </div>
          <nav id="navigation" class="menu">
            <ul class="responsive float-right">
              <template v-if="!isLoggedIn">
                <li>
                  <router-link to="/kayit-ol">
                    <i class="fa fa-user"></i>
                    Kayıt Ol
                  </router-link>
                </li>
                <li>
                  <router-link to="/giris-yap">
                    <i class="fa fa-lock"></i>
                    Giriş Yap
                  </router-link>
                </li>
              </template>
              <li v-if="isLoggedIn">
                <a href="" @click="logout">Çıkış Yap</a>
              </li>
              <li class="header-subs">
                <router-link to="/abone-ol">
                  <i class="fa fa-bullhorn" />
                  <span class>
                    Abone ol!
                  </span>
                </router-link>
              </li>
              <li>
                <router-link class="add-post button tag-create-post" to="/ilan-ekle">
                  <i class="fa fa-plus" />
                  <span class="tag-create-post">
                    İlan Ekle
                  </span>
                </router-link>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <div class="clearfix" />
    <app-banner v-if="withBanner" />
  </section>
</template>

<style lang="scss">
.app-header {
  #logo h2 {
    line-height: 98px;
    font-size: 36px;

    a {
      color: #333;

      span {
        color: #26ae61;
        font-weight: 500;
      }
    }
  }

  #navigation {
    a {
      background: #26ae61;
      color: #fff;
      font-size: 16px;
      padding: 12px 22px;
    }

    .header-subs a {
      background: #666;
    }

    i {
      margin-right: 10px !important;
    }
  }
}
</style>
