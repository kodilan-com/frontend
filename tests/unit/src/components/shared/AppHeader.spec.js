import { shallowMount } from '@vue/test-utils';
import AppHeader from '@/components/shared/AppHeader';
import AppBanner from '@/components/shared/AppBanner';

describe('AppBanner', () => {
  const generateWrapper = (propsData = {}) => shallowMount(AppHeader, {
    propsData,
    stubs: [
      'router-link',
    ],
  });
  describe('props', () => {
    it('should render AppBanner component if withBanner props is given as true', () => {
      const wrapper = generateWrapper({
        withBanner: true,
      });

      expect(wrapper.find(AppBanner).exists()).toBe(true);
    });

    it('should not render AppBanner component if withBanner props is given as false', () => {
      const wrapper = generateWrapper({
        withBanner: false,
      });

      expect(wrapper.find(AppBanner).exists()).toBe(false);
    });


    it('should not render AppBanner component if withBanner props is not given', () => {
      const wrapper = generateWrapper();

      expect(wrapper.find(AppBanner).exists()).toBe(false);
    });
  });

  describe('template', () => {
    it('should match the snapshot', () => {
      const wrapper = generateWrapper();

      expect(wrapper.vm.$el).toMatchSnapshot();
    });
  });
});
