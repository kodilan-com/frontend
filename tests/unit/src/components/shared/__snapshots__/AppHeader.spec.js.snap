// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBanner.vue template should match the snapshot 1`] = `
<section>
  <app-topbar-stub />
   
  <header
    class="app-header sticky-header"
  >
    <div
      class="container"
    >
      <div
        class="sixteen columns"
      >
        <div
          id="logo"
        >
          <h2>
            <router-link-stub
              to="/"
            >
              <span>
                { 
              </span>
              kod
              <span>
                , 
              </span>
              ilan
              <span>
                 }
              </span>
            </router-link-stub>
          </h2>
        </div>
         
        <nav
          class="menu"
          id="navigation"
        >
          <ul
            class="responsive float-right"
          >
            <li
              class="header-subs"
            >
              <router-link-stub
                to="/abone-ol"
              >
                <i
                  class="fa fa-bullhorn"
                />
                 
                <span>
                  
                  Abone ol!
                
                </span>
              </router-link-stub>
            </li>
             
            <li>
              <router-link-stub
                class="add-post button tag-create-post"
                to="/ilan-ekle"
              >
                <i
                  class="fa fa-plus"
                />
                 
                <span
                  class="tag-create-post"
                >
                  
                  Ücretsiz İlan Ekle
                
                </span>
              </router-link-stub>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
   
  <div
    class="clearfix"
  />
   
  <!---->
</section>
`;
